File: santerhivyduval-duval_and_ivy/PlanRhApp/src/app/models/planning.ts
export interface Planning {
  _id?: string;
  user_id: string;
  date: string;
  activity_code: 'SOIN' | 'CONGÉ' | 'REPOS' | 'FORMATION' | 'ADMINISTRATIF';
  plage_horaire: string;
  created_at?: string;
  updated_at?: string;
  validated_by?: string;
  commentaire?: string;
  user_name?: string;
  user_matricule?: string;
}

File: santerhivyduval-duval_and_ivy/PlanRHAPI/schemas/planning.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, time, datetime

class PlanningCreate(BaseModel):
    user_id: str
    date: str  # Format: YYYY-MM-DD
    activity_code: str  # "SOIN", "CONGÉ", "REPOS", etc.
    plage_horaire: str  # Format: "HH:MM-HH:MM" ou description
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    validated_by: Optional[str] = None  # ID du cadre qui a validé
    commentaire: Optional[str] = None

class PlanningUpdate(BaseModel):
    activity_code: Optional[str] = None
    plage_horaire: Optional[str] = None
    commentaire: Optional[str] = None
    updated_at: Optional[datetime] = None

class PlanningResponse(BaseModel):
    id: str
    user_id: str
    date: str
    activity_code: str
    plage_horaire: str
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    validated_by: Optional[str] = None
    commentaire: Optional[str] = None
    user_name: Optional[str] = None
    user_matricule: Optional[str] = None








